pkg/agent/config/config.go

```diff
@@ -47,13 +51,33 @@ func Get(ctx context.Context, agent cmds.Agent, proxy proxy.Proxy) *config.Node
 			case <-time.After(5 * time.Second):
 				continue
 			case <-ctx.Done():
-				logrus.Fatalf("Interrupted")
+				logrus.Fatal("Interrupted")
 			}
 		}
 		return agentConfig
 	}
 }
 
+// GetKubeProxyDisabled returns true if kube-proxy has been disabled in the server configuration.
+// The server may not have a complete view of cluster configuration until after all startup
+// hooks have completed, so a call to this will block until after the server's readyz endpoint
+// returns OK.
+func GetKubeProxyDisabled(ctx context.Context, node *config.Node, proxy proxy.Proxy) bool {
+	for {
+		disabled, err := getKubeProxyDisabled(ctx, node, proxy)
+		if err != nil {
+			logrus.Errorf("Failed to configure kube-proxy: %v", err)
+			select {
+			case <-time.After(5 * time.Second):
```

Comment:
The common convention here is to do the what's below rather than a time.After.

```go
ticker := time.NewTicker(5 * time.Second)
defer ticker.Stop() 

for range ticker.C {
    ...
```

Reply:
Should we just drop the `ctx.Done()` case entirely then?

Reply:
That'd depend on the context. If it's one with a cancel, I'd say no since it could receive a signal for cancellation from somewhere. If it's not, then yes.

Reply:
Also, if I change the outer `for` to range on the ticker, instead of selecting inside the loop, won't it always wait 5 seconds before trying the first time?

Reply:
It's the main CLI context